JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToGR/UniqueToGR.tsx

```diff
@@ -15,34 +15,39 @@ import SaveIcon from '@mui/icons-material/Save'
 import CancelIcon from '@mui/icons-material/Close'
 import { useEffect, useState } from 'react'
 import { EditToolbar } from 'components/shared/EditToolBar'
-import { transformFieldName } from 'utils/helpers'
-
-const UniqueToGR = ({
-  uniqueToGoldenRecordData
-}: {
-  uniqueToGoldenRecordData: any
-}) => {
-  const [rows, setRows] = useState(uniqueToGoldenRecordData)
+import { toSnakeCase, transformFieldName } from 'utils/helpers'
+import { useConfiguration } from 'hooks/useUIConfiguration'
+import { Configuration, Field } from 'types/Configuration'
+
+const UniqueToGR = () => {
+  const [rows, setRows] = useState<any>([])
   const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({})
+  const {configuration, setConfiguration} = useConfiguration();
 
   useEffect(() => {
-    const rowsWithIds = uniqueToGoldenRecordData.map(
-      (row: any, index: number) => ({
-        ...row,
-        id: index.toString()
-      })
-    )
-    setRows(rowsWithIds)
-  }, [uniqueToGoldenRecordData])
+    if(configuration && configuration.auxGoldenRecordFields){
+      const rowData = configuration.auxGoldenRecordFields.map(
+        (row: any, rowIndex: number) => ({
+          id: rowIndex + 1,
+          ...row,
+          rowIndex
+        })
+      )
+      setRows(rowData)
+    }
+  
+  }, [configuration])
 
   const handleEditClick = (id: GridRowId) => () => {
     setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.Edit } })
   }
 
   const handleSaveClick = (id: GridRowId) => () => {
-    setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } })
+    const updatedRow = rows.find((row: { id: GridRowId }) => row.id === id)
+    if (updatedRow) {
+      setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } })
```

Comment:
@NyashaMuusha Same question here. Just double checking if/why this second call is needed...